### PR TITLE
Getting Started Docs

### LINKED ISSUE #146: Add "Getting Started" section to docs
* read/write API
* basic examples
* maybe a section on why JSON3 vs others
* anything else? (@quinnj )

### PR DESCRIPTION
**Pull request by mcmcgrath13**
* Add getting started to docs
* Doc strings for the main API
* Examples to highlight main features

I took my best guess at the docstrings and kwargs based on the code, tests, and past issues & prs.  I'm sure there are things that I missed or could use a better explanation.  I wasn't sure what to put for why JSON3 vs others, but I think that would be a helpful section (or maybe even just pointers to other JSON-related packages).

closes #146

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -4,6 +4,147 @@
...
+hello_world = JSON3.read(json_string)
+
+open("my_new_file.json", "w") do io
+    JSON3.write(io, hello_world)
+end
+```
+
+#### Read JSON into a type
+
+See more details on the types that are provided and how to customize parsing [below](#Builtin-types).
```
Comment (quinnj) [inline docs/src/index.md:51]: I think this should link to `## Struct API` instead?

Comment (mcmcgrath13) [reply docs/src/index.md:51]: I linked it to the builtin-types as that serves as in intro for what will happen automagically vs when struct mapping is needed, but I agree that given this example context most will likely just want to jump straight to `## Struct API`.

File: src/read.jl [hunk 2]
```diff
@@ -9,6 +9,23 @@ Base.codeunits(x::VectorString) = x.bytes
...
+
+## Args
+
+* `json_str`: A string, IO, or bytes (`AbstractVector{UInt8`) containing JSON to read
+* `type`: Optionally, a type to read the JSON into. If not a [built in type](#Builtin-types), must have a "struct mapping" registered with [StructTypes.jl](#Struct-API).
+
+## Keyword Args
+
+* `jsonlines`: A Bool indicating that the `json_str` contains newline delimited JSON strings, which will be read into a `JSON3.Array` of the JSON values.  See [jsonlines](https://jsonlines.org/) for reference. [default `false`]
+* `allow_inf`: Allow reading of `Inf` values (not part of the JSON standard). [default `false`]
```
[the remark this answers is not in the document]
Comment (mcmcgrath13) [reply src/read.jl]: Should this be changed on `write` as well?

[the remark this answers is not in the document]
Comment (quinnj) [reply src/read.jl]: Yes

**Comment by mcmcgrath13** [comment]
@quinnj should `pretty(out::IO, x) = pretty(out, JSON3.write(x))` also take the `kw...` args? (I've included this in the latest commit, probably worth a closer look)

**Review by quinnj — APPROVED** [review]

**Pull request merged**